client/signup/steps/page-picker/shopping-cart-for-difm.tsx

```diff
@@ -0,0 +1,196 @@
+	}
+`;
+const DummyLineItemContainer = styled.div`
+	display: flex;
+	flex-wrap: wrap;
+	justify-content: space-between;
+	font-weight: 400;
+	color: #2c3338;
+	font-size: 14px;
+	padding: 16px 0;
+	border-bottom: 1px solid #dcdcde;
+	position: relative;
+	.page-picker__title {
+		flex: 1;
+		word-break: break-word;
+		@media ( max-width: 600px ) {
+			display: flex;
+			font-size: 0.9em;
+		}
+	}
+	.page-picker__price {
+		font-weight: 500;
+	}
+	.page-picker__meta {
+		color: #646970;
+		font-size: 12px;
+		width: 100%;
+		display: flex;
+		flex-direction: row;
+		align-content: center;
+		justify-content: space-between;
+		flex-wrap: wrap;
+		gap: 2px 10px;
+	}
+	span {
+		position: relative;
+	}
+
+	@media ( max-width: 600px ) {
+		font-size: 0.9em;
+		padding: 2px 0;
+		border: none;
+
+		.page-picker__meta {
+			display: none;
+		}
+	}
+`;
+
+const LineItemsWrapper = styled.div`
+	box-sizing: border-box;
+	margin: 20px 0;
+	padding: 0;
+	overflow-y: auto;
+	max-height: 75vh;
+	width: 100%;
+	max-width: 365px;
+	border-bottom: 1px solid #dcdcde;
+	@media ( max-width: 600px ) {
+		margin: 0 0px;
+		border: none;
+		max-width: 100%;
+	}
+`;
+
+const Total = styled.div`
+	font-weight: 500;
+	display: flex;
+	flex-wrap: wrap;
+	justify-content: space-between;
+	padding: 30px 0;
+	div {
+		display: flex;
+		align-items: center;
+		&.value {
+			font-family: Recoleta;
+			font-size: 32px;
+			font-weight: 400;
+			@media ( max-width: 600px ) {
+				font-size: 16px;
+			}
+		}
+	}
+
+	@media ( max-width: 600px ) {
+		font-size: 16px;
+		margin-top: 6px;
+		padding: 0;
+		padding-top: 6px;
+		border-top: 1px solid #eee;
+	}
+`;
+
+function DummyLineItem( { product, meta, productCount, nameOverride }: CartItem ) {
+	const currencyCode = useSelector( getCurrentUserCurrencyCode ) as string;
+
+	if ( ! product ) {
+		return null;
+	}
+	return (
+		<DummyLineItemContainer>
+			<div className="page-picker__title">{ nameOverride ?? product.product_name }</div>
+			<div className="page-picker__price">
+				{ productCount !== undefined
+					? formatCurrency( product.cost * productCount, currencyCode, { precision: 0 } )
+					: formatCurrency( product.cost, currencyCode, { precision: 0 } ) }
+			</div>
+			{ meta && <div className="page-picker__meta">{ meta }</div> }
+		</DummyLineItemContainer>
+	);
+}
+
+export default function ShoppingCartForDIFM( { selectedPages }: { selectedPages: string[] } ) {
+	const translate = useTranslate();
+	const { items, total, isCartLoading } = useCartForDIFM( selectedPages );
+	const signupDependencies = useSelector( getSignupDependencyStore );
+	const { newOrExistingSiteChoice } = signupDependencies;
+
+	return newOrExistingSiteChoice === 'existing-site' && isCartLoading ? (
+		<LoadingContainer>
+			{ items.length === 0 ? (
+				<>
+					<LoadingLine key="plan-placehorder" />
+					<LoadingLine key="difm-placehorder" />
+				</>
+			) : (
+				items.map( ( lineItem, i ) => <LoadingLine key={ i } /> )
```

Comment:
Using an array index as a React key is a poor idea unless there is no other option because the array index may change and cause React to render the elements incorrectly (it uses the index to identify which element is which).

In this case, we have the `lineItem` object that might have some unique identifying information. How about the product name?

```suggestion
				items.map( ( lineItem ) => <LoadingLine key={  lineItem.product.product_name } /> )
```

Reply:
Ah yes! Completely agree, I sort of ignored this best practice because it was just the load screen . I will/have fixed this!